tskv/src/wal/wal_store.rs

```diff
@@ -438,39 +458,42 @@ impl RaftEntryStorageInner {
     async fn recover_record(
         &mut self,
         wal_id: u64,
-        record: Record,
+        record: WalRecordData,
         apply_id: Option<LogId<u64>>,
         vnode_store: &mut VnodeStorage,
     ) -> TskvResult<()> {
-        if record.data.len() < 9 {
-            return Ok(());
-        }
-
-        let wal_entry = WalRecordData::new(record.data);
-
-        if let Some(entry) = wal_entry.block {
-            if let Some(apply_id) = apply_id {
-                let last_seq = vnode_store.ts_family().read().await.version().last_seq();
-                if entry.log_id.index > last_seq && entry.log_id.index <= apply_id.index {
-                    if let EntryPayload::Normal(ref req) = entry.payload {
-                        let ctx = replication::ApplyContext {
-                            index: entry.log_id.index,
-                            raft_id: self.wal.vnode_id as u64,
-                            apply_type: replication::APPLY_TYPE_WAL,
-                        };
-
-                        let request = parse_prost_bytes::<RaftWriteCommand>(req)
-                            .map_err(|e| DecodeSnafu.into_error(Box::new(e)))?;
-                        if let Some(command) = request.command {
-                            vnode_store.apply(&ctx, command).await?;
-                        }
+        let entry = record.block;
+        if let Some(apply_id) = apply_id {
+            let last_seq = vnode_store.ts_family().read().await.version().last_seq();
+            if entry.log_id.index > last_seq && entry.log_id.index <= apply_id.index {
+                if let EntryPayload::Normal(ref req) = entry.payload {
+                    let ctx = replication::ApplyContext {
+                        index: entry.log_id.index,
+                        raft_id: self.wal.vnode_id as u64,
+                        apply_type: replication::APPLY_TYPE_WAL,
+                    };
+
+                    let request = parse_prost_bytes::<RaftWriteCommand>(req)
+                        .map_err(|e| DecodeSnafu.into_error(Box::new(e)))?;
+                    if let Some(command) = request.command {
+                        vnode_store.apply(&ctx, command).await?;
                     }
                 }
             }
 
-            self.mark_write_wal(entry, wal_id, record.pos).await?;
+            if entry.log_id.index > apply_id.index {
```

Comment:
why truncate not applied wal entries？

Reply:
Unapplied wal during restart will be truncated and will not be applied again